Homework 5 Avatars

marisereda opened the pull request.

marisereda committed:
feat:add avatar to user model and update avatar controller

marisereda committed:
feat: add unit-test for signup controller

marisereda committed:
fix: fix updateAvatar controller

Harryk13 commented on controllers/users/updateAvatar.js:
```diff
@@ -0,0 +1,26 @@
+const updateAvatar = async (req, res) => {
+  const { _id: userId } = req.user;
+  const { filename, path: tempUpload } = req.file;
+  const avatarPath = path.resolve("public", "avatars", filename);
```
а якщо вдруге завантажить хтось аватарт з таким самим ім'ям?

marisereda replied:
Ім'я файла - унікальне, формується у middleware "upload.js": до імені файла додається userId.

```js
const multerConfig = multer.diskStorage({
  destination: (req, file, cb) => {
    cb(null, tempDir);
  },
  filename: (req, file, cb) => {
    const { _id: userId } = req.user;
    const fileName = `${userId}_${file.originalname}`;
    cb(null, fileName);
  },
  limits: {
    fileSize: 2048,
  },
});
```

Harryk13 replied:
ну так ось, той самий юзер завантажує ще один аватар з таким самим ім'ям і ваше ім'я вже не унікальне

marisereda replied:
У комміті "fix: make unique avatar file name":
- змінено формування назви файлу: тепер назва - uuid
- додано фільтр типу файла (лише image)  до multer
- якщо користувач завантажує новий аватар, файл із старим аватаром видаляється

Harryk13 replied:
Супер

marisereda committed:
fix: make unique avatar file name

- add filter for avatar file type (only image files) to multer
- avatar file name is generated with uuid
- when user add new avatar file the old one is removed
